plugins/GSdx/GSLzma.cpp

```diff
@@ -162,19 +162,17 @@ bool GSDumpRaw::IsEof() {
 	return feof(m_fp);
 }
 
-void GSDumpRaw::Read(void* ptr, size_t size) {
-	if (size == 1) {
-		// I don't know why but read of size 1 is not happy
-		int v = fgetc(m_fp);
-		memcpy(ptr, &v, 1);
-	} else {
-		size_t ret = fread(ptr, 1, size, m_fp);
-		if (ret != size) {
-			fprintf(stderr, "GSDumpRaw:: Read error (%zu/%zu)\n", ret, size);
-			throw "BAD"; // Just exit the program
-		}
+bool GSDumpRaw::Read(void* ptr, size_t size) {
+	size_t ret = fread(ptr, 1, size, m_fp);
+	if (ret != size && ferror(m_fp)) {
```

Comment:
> It makes more sense to use || over here. What about cases when only one of the condition is false? you won't be throwing the error.

That's intended behaviour. If an n-byte `fread` reads less than n bytes, then the EOF might have been reached, which is not an error.

> I'm not sure at which scenarios `ferror()` returns 1, reference at some sites just mentions that 1 will be set when the previous file operation has failed but it's still not fully clear on how they define a failed operation. Plus C library code for ferror() doesn't make sense to me.

You need to look at `fread`, `fwrite`, `fseek`, etc., not at `ferror`. I'm not sure about what doesn't make sense to you in the glibc ferror code.

Side note: cppreference is the better reference site IMO.